### PR TITLE
sql: update generated columns docs

### PR DESCRIPTION
**Pull request by crazycs520**
Currently only generated `stored` column index take effective, create index on generated `virtual` column is useless for plan-optimizer.

Below is example, there is 2 tables: `persone`, `persone1`. The only differented between `persone` and `persone1` is the column `city` virtual or stored generated column.
```sql
tidb:test> show create table person
+--------+----------------------------------------------------------------------------------------------------------+
| Table  | Create Table                                                                                             |
+--------+----------------------------------------------------------------------------------------------------------+
| person | CREATE TABLE `person` (                                                                                  |
|        |   `id` int(11) NOT NULL AUTO_INCREMENT,                                                                  |
|        |   `name` varchar(255) NOT NULL,                                                                          |
|        |   `address_info` json DEFAULT NULL,                                                                      |
|        |   `city` varchar(64) GENERATED ALWAYS AS (json_unquote(json_extract(`address_info`, "$.city"))) VIRTUAL, |
|        |   PRIMARY KEY (`id`),                                                                                    |
|        |   KEY `city` (`city`)                                                                                    |
|        | ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin AUTO_INCREMENT=30002                               |
+--------+----------------------------------------------------------------------------------------------------------+
1 row in set
Time: 0.012s
tidb:test> show create table person1
+---------+---------------------------------------------------------------------------------------------------------+
| Table   | Create Table                                                                                            |
+---------+---------------------------------------------------------------------------------------------------------+
| person1 | CREATE TABLE `person1` (                                                                                |
|         |   `id` int(11) NOT NULL AUTO_INCREMENT,                                                                 |
|         |   `name` varchar(255) NOT NULL,                                                                         |
|         |   `address_info` json DEFAULT NULL,                                                                     |
|         |   `city` varchar(64) GENERATED ALWAYS AS (json_unquote(json_extract(`address_info`, "$.city"))) STORED, |
|         |   PRIMARY KEY (`id`),                                                                                   |
|         |   KEY `city` (`city`)                                                                                   |
|         | ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin AUTO_INCREMENT=30002                              |
+---------+---------------------------------------------------------------------------------------------------------+
1 row in set
Time: 0.006s
tidb:test> select * from person where city="beijing"
+----+------+---------------------+---------+
| id | name | address_info        | city    |
+----+------+---------------------+---------+
| 1  | a    | {"city": "beijing"} | beijing |
+----+------+---------------------+---------+
1 row in set
Time: 0.014s
tidb:test> select * from person1 where city="beijing"
+----+------+---------------------+---------+
| id | name | address_info        | city    |
+----+------+---------------------+---------+
| 1  | a    | {"city": "beijing"} | beijing |
+----+------+---------------------+---------+
1 row in set
tidb:test> desc select * from person where city="beijing"
+--------------------+-------+------+----------------------------------------------------------------------------------------------------------------------------------+
| id                 | count | task | operator info                                                                                                                    |
+--------------------+-------+------+----------------------------------------------------------------------------------------------------------------------------------+
| Projection_7       | 0.80  | root | test.person.id, test.person.name, test.person.address_info, cast(json_unquote(json_extract(test.person.address_info, "$.city"))) |
| └─Selection_8      | 0.80  | root | eq(cast(json_unquote(json_extract(test.person.address_info, "$.city"))), "beijing")                                              |
|   └─TableReader_10 | 1.00  | root | data:TableScan_9                                                                                                                 |
|     └─TableScan_9  | 1.00  | cop  | table:person, range:[-inf,+inf], keep order:false, stats:pseudo                                                                  |
+--------------------+-------+------+----------------------------------------------------------------------------------------------------------------------------------+
4 rows in set
Time: 0.010s
tidb:test> desc select * from person1 where city="beijing"
+----------------+-------+------+----------------------------------------------------------------------------------------+
| id             | count | task | operator info                                                                          |
+----------------+-------+------+----------------------------------------------------------------------------------------+
| IndexLookUp_10 | 0.00  | root |                                                                                        |
| ├─IndexScan_8  | 0.00  | cop  | table:person1, index:city, range:["beijing","beijing"], keep order:false, stats:pseudo |
| └─TableScan_9  | 0.00  | cop  | table:person1, keep order:false, stats:pseudo                                          |
+----------------+-------+------+----------------------------------------------------------------------------------------+
```

### CODE REVIEW

**Comment by crazycs520** [comment]
@zz-jason PTAL

File: sql/generated-columns.md [hunk 1]
```diff
@@ -21,21 +21,21 @@ CREATE TABLE person (
 );
 ```
 
-为 JSON 列添加索引之前，首先必须抽取该列为 generated column。以 `city` generated column 为例，你可以添加索引：
+为 JSON 列添加索引之前，首先必须抽取该列为 generated stored column。注意：必须是 generated stored column 上建立的索引才能被优化器使用到, 如果在 generated virtual column 上建立索引，优化器将无法使用这个索引。下面以 `city` generated stored column 为例，你可以添加索引：
```
Comment (shenli) [inline sql/generated-columns.md]: “优化器将无法使用这个索引” -> "优化器目前将无法使用这个索引，会在后续版本中改进"

**Comment by morgo** [comment]
May I suggest we reference the issue in the docs? https://github.com/pingcap/tidb/issues/5189

This will help us understand the docs are out of date when the issue is closed.

File: sql/generated-columns.md (hunk 1 above)
Comment (crazycs520) [reply sql/generated-columns.md]: Done. And add related issue link in doc.

**Comment by crazycs520** [comment]
@morgo , Yes, I add the issue link  in the doc.  Thanks.

**Comment by lilin90** [comment]
@CaitinChen There are multiple format issues in this PR. Could you please help fix them in another PR? Thanks!

**Comment by CaitinChen** [comment]
@lilin90 OK.

**Pull request merged**